src/DateTimeInput/index.jsx

```diff
@@ -426,310 +466,215 @@ export default class DateTimeInput extends Component
         } );
 
         return _.chunk( months, 4 );
-    }
-
-    monthLabel()
-    {
-        const month = $m( this.state.gridStartTimestamp ).month();
-        return copy.months[ month ];
-    }
+    };
 
-    yearLabel()
-    {
-        return $m( this.state.gridStartTimestamp ).year().toString();
-    }
 
-    handleClickCell( { value } )
+    const monthLabel = () =>
     {
-        const { isReadOnly } = this.props;
+        const month = $m( gridStartTimestamp ).month();
+        return copy.months[ month ];
+    };
 
-        if ( !isReadOnly )
-        {
-            this.setState( { timestamp: value } );
-            const { onChange } = this.props;
-            const { id } = this.state;
-            if ( typeof onChange === 'function' )
-            {
-                onChange( { id, value } );
-            }
-        }
+    const yearLabel = () => $m( gridStartTimestamp ).year().toString();
 
-        this.purgeEdits();
-    }
 
-    purgeEdits()
+    const purgeEdits = () =>
```

Comment:
I guess stuff like this can use `useCallback` with an empty array:

```
const purgeEdits = useCallback( 
    () => {
        setEditingHourInputValue( undefined );
        setEditingMainInputValue( undefined );
        setEditingMinuteInputValue( undefined );   
     }, 
    [],
):
```

This way we don’t redefine the function on every render

Reply:
Same applies for `open()`, `close()`

Reply:
done 👍